provider/pkg/watcher/watcher.go

```diff
@@ -203,7 +203,3 @@ func RetryableError(err error) *RetryError {
 	}
 	return &RetryError{Err: err, Retryable: true}
 }
-
```

Comment:
No, the seed is randomly initialized now when the program starts up, so `Seed` is only needed for determinism (e.g., tests). See https://github.com/golang/go/issues/56319 for more info.

Reply:
Oh, I just realized that the release workflow is still using Go 1.19, so this actually wouldn't work. I'll update the workflows separately, and then it will work as expected:
https://pkg.go.dev/math/rand@go1.20.0#Seed

Reply:
Updated the build to use Go v1.20